public final class libcore.io.GaiException extends java.lang.RuntimeException
{
    public final int error;
    private final java.lang.String functionName;

    public void <init>(java.lang.String, int)
    {
        libcore.io.GaiException r0;
        java.lang.String r1;
        int i0;

        r0 := @this: libcore.io.GaiException;

        r1 := @parameter0: java.lang.String;

        i0 := @parameter1: int;

        specialinvoke r0.<java.lang.RuntimeException: void <init>()>();

        r0.<libcore.io.GaiException: java.lang.String functionName> = r1;

        r0.<libcore.io.GaiException: int error> = i0;

        return;
    }

    public void <init>(java.lang.String, int, java.lang.Throwable)
    {
        libcore.io.GaiException r0;
        java.lang.String r1;
        int i0;
        java.lang.Throwable r2;

        r0 := @this: libcore.io.GaiException;

        r1 := @parameter0: java.lang.String;

        i0 := @parameter1: int;

        r2 := @parameter2: java.lang.Throwable;

        specialinvoke r0.<java.lang.RuntimeException: void <init>(java.lang.Throwable)>(r2);

        r0.<libcore.io.GaiException: java.lang.String functionName> = r1;

        r0.<libcore.io.GaiException: int error> = i0;

        return;
    }

    public java.lang.String getMessage()
    {
        libcore.io.GaiException r0;
        java.lang.String r1, r2, $r5, $r13;
        int $i0, $i1, $i2;
        libcore.io.Os $r3;
        java.lang.StringBuilder $r4, $r6, $r7, $r8, $r9, $r10, $r11, $r12, $r14, $r15;

        r0 := @this: libcore.io.GaiException;

        $i0 = r0.<libcore.io.GaiException: int error>;

        r1 = staticinvoke <libcore.io.OsConstants: java.lang.String gaiName(int)>($i0);

        if r1 != null goto label1;

        $r12 = new java.lang.StringBuilder;

        specialinvoke $r12.<java.lang.StringBuilder: void <init>()>();

        $r14 = virtualinvoke $r12.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>("GAI_ error ");

        $i2 = r0.<libcore.io.GaiException: int error>;

        $r15 = virtualinvoke $r14.<java.lang.StringBuilder: java.lang.StringBuilder append(int)>($i2);

        r1 = virtualinvoke $r15.<java.lang.StringBuilder: java.lang.String toString()>();

     label1:
        $r3 = <libcore.io.Libcore: libcore.io.Os os>;

        $i1 = r0.<libcore.io.GaiException: int error>;

        r2 = interfaceinvoke $r3.<libcore.io.Os: java.lang.String gai_strerror(int)>($i1);

        $r4 = new java.lang.StringBuilder;

        specialinvoke $r4.<java.lang.StringBuilder: void <init>()>();

        $r5 = r0.<libcore.io.GaiException: java.lang.String functionName>;

        $r6 = virtualinvoke $r4.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>($r5);

        $r7 = virtualinvoke $r6.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(" failed: ");

        $r8 = virtualinvoke $r7.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(r1);

        $r9 = virtualinvoke $r8.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(" (");

        $r10 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(r2);

        $r11 = virtualinvoke $r10.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(")");

        $r13 = virtualinvoke $r11.<java.lang.StringBuilder: java.lang.String toString()>();

        return $r13;
    }

    public java.net.UnknownHostException rethrowAsUnknownHostException() throws java.net.UnknownHostException
    {
        libcore.io.GaiException r0;
        java.lang.String $r1;
        java.net.UnknownHostException $r2;

        r0 := @this: libcore.io.GaiException;

        $r1 = virtualinvoke r0.<libcore.io.GaiException: java.lang.String getMessage()>();

        $r2 = virtualinvoke r0.<libcore.io.GaiException: java.net.UnknownHostException rethrowAsUnknownHostException(java.lang.String)>($r1);

        throw $r2;
    }

    public java.net.UnknownHostException rethrowAsUnknownHostException(java.lang.String) throws java.net.UnknownHostException
    {
        libcore.io.GaiException r0;
        java.lang.String r1;
        java.net.UnknownHostException r2, $r3;

        r0 := @this: libcore.io.GaiException;

        r1 := @parameter0: java.lang.String;

        $r3 = new java.net.UnknownHostException;

        specialinvoke $r3.<java.net.UnknownHostException: void <init>(java.lang.String)>(r1);

        r2 = $r3;

        virtualinvoke r2.<java.net.UnknownHostException: java.lang.Throwable initCause(java.lang.Throwable)>(r0);

        throw r2;
    }
}
